### PR TITLE
[OpenAI] Add azure-json dependency to azure-ai-openai

### PR DESCRIPTION
**Pull request by mssfang**
- Make the module azure-json dependency not required for transitive dependencies

- (azure-ai-openai only) Added the missing azure-json dependency for compile time

### CODE REVIEW

File: sdk/openai/azure-ai-openai-assistants/src/main/java/module-info.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 module com.azure.ai.openai.assistants {
     requires transitive com.azure.core;
-    requires transitive com.azure.json;
+    requires com.azure.json;
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai-assistants/src/main/java/module-info.java:7]: We should keep this transitive.

File: sdk/openai/azure-ai-openai/pom.xml [hunk 2]
```diff
@@ -62,6 +62,11 @@
       <artifactId>azure-core</artifactId>
       <version>1.51.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-json</artifactId>
+      <version>1.2.0</version> <!-- {x-version-update;com.azure:azure-json;dependency} -->
+    </dependency>
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/pom.xml:69]: What is the reason for adding this direct dependency?

File: sdk/openai/azure-ai-openai/src/main/java/module-info.java [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 module com.azure.ai.openai {
     requires transitive com.azure.core;
-    requires transitive com.azure.json;
+    requires com.azure.json;
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/module-info.java:7]: Same here - we should keep this transitive.

File: sdk/openai/azure-ai-openai-assistants/src/main/java/module-info.java (hunk 1 above)
Comment (mssfang) [reply sdk/openai/azure-ai-openai-assistants/src/main/java/module-info.java:7]: From the migration guideline, we don't require the transitive.

File: sdk/openai/azure-ai-openai/pom.xml (hunk 2 above)
Comment (mssfang) [reply sdk/openai/azure-ai-openai/pom.xml:69]: It is from the migration guideline. https://github.com/Azure/azure-sdk-for-java/wiki/Azure-Json-Migration#update-project-configurations

Comment (srnagar) [reply sdk/openai/azure-ai-openai/pom.xml:69]: Adding the dependency is fine but we don't need to change anything in module-info.java. Azure Core already requires azure-json transitively.

File: sdk/openai/azure-ai-openai-assistants/src/main/java/module-info.java (hunk 1 above)
Comment (srnagar) [reply sdk/openai/azure-ai-openai-assistants/src/main/java/module-info.java:7]: This is not needed as azure-core is already doing this. We should update the migration guide.

File: sdk/openai/azure-ai-openai/pom.xml (hunk 2 above)
Comment (mssfang) [reply sdk/openai/azure-ai-openai/pom.xml:69]: If it is not a requirement, i would rather to leave it as it is and close this PR.

**Pull request closed without merging**